src/main/java/com/flagsmith/flagengine/utils/types/TypeCasting.java

```diff
@@ -115,8 +144,32 @@ public static Boolean toBoolean(Object str) {
    * @return
    */
   public static Boolean isBoolean(Object str) {
-    String value = ((String) str).toLowerCase();
-    return Boolean.TRUE.toString().toLowerCase().equals(value)
-        || Boolean.FALSE.toString().toLowerCase().equals(value);
+    return str instanceof Boolean
+        || Boolean.TRUE.toString().equalsIgnoreCase(((String) str))
+        || Boolean.FALSE.toString().equalsIgnoreCase(((String) str));
+  }
+
+  /**
+   * Convert the object to Semver.
+   * @param str Object to convert to Semver.
+   * @return
+   */
+  public static ComparableVersion toSemver(Object str) {
+    try {
+      String value = SemanticVersioning.isSemver((String) str)
+          ? SemanticVersioning.removeSemver((String) str) : ((String) str);
+      return new ComparableVersion(value);
+    } catch (Exception nfe) {
```

Comment:
Again, why catch generic `Exception`?

Reply:
Then shouldn't we just catch both `NumberFormatException` and `RuntimeException` rather than `Exception` ?

Reply:
Resolving as I've spun this off into a separate issue [here](https://github.com/Flagsmith/flagsmith-java-client/issues/70).